src/runtime_src/core/edge/user/device_linux.cpp

```diff
@@ -55,6 +56,22 @@ using key_type = query::key_type;
 xclDeviceHandle handle;
 xclDeviceInfo2 deviceInfo;
 
+struct drm_fd
+{
+  int fd;
+  drm_fd(const std::string file_path, int flags)
```

Comment:
`const std::string& `  Always pass 'complex' objects by const reference
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-in